### PR TITLE
Remove two stage constuction

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

**Comment by matklad** [comment]
bors r+
🤖

File: editors/code/src/ctx.ts [hunk 1]
```diff
@@ -15,29 +14,21 @@ export class Ctx {
...
+    static async create(config: Config, extCtx: vscode.ExtensionContext, serverPath: string): Promise<Ctx> {
+        const client = await createClient(config, serverPath);
+        const res = new Ctx(config, extCtx, client);
+        res.pushCleanup(client.start());
         await client.onReady();
+        return res;
+    }
 
-        this.client = client;
+    private constructor(config: Config, extCtx: vscode.ExtensionContext, client: lc.LanguageClient) {
```
Comment (Veetaha) [inline editors/code/src/ctx.ts]: @matklad , what about shortening the code?
```
private constructor(
    private readonly config: Config,
    private readonly extCtx: vscode.ExtensionContextx,
    private client: lc.LanguageClient
) {}
```

Comment (matklad) [reply editors/code/src/ctx.ts]: wow, didn't realize that TypeScript can do that, thanks!

**Pull request merged**